src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/issuefinaldecision/IssueFinalDecisionMidEventHandler.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.issuefinaldecision;
+
+import static java.util.Collections.singletonList;
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.isYes;
+
+import java.util.List;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+public class IssueFinalDecisionMidEventHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    private static final List<String> DEATH_OF_APPELLANT_WARNING_PAGES = singletonList("previewDecisionNotice");
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.MID_EVENT)
+                && callback.getEvent() == EventType.ISSUE_FINAL_DECISION;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        if (isYes(sscsCaseData.getIsAppellantDeceased()) && DEATH_OF_APPELLANT_WARNING_PAGES.contains(callback.getPageId())) {
```

Comment:
I'm not sure this will work? Don't you need to ignore the warning second time round e.g. `!callback.isIgnoreWarnings()`? Otherwise, they'll get stuck in a loop.

Reply:
It did when I tried it locally. Perhaps it won't work in the AboutToSubmit callback. There should be no harm to add this in.